oci/repositories.bzl

```diff
@@ -146,23 +165,27 @@ def oci_register_toolchains(name, crane_version, zot_version, register = True):
             crane_version = crane_version,
         )
 
-        zot_repositories(
-            name = "{name}_zot_{platform}".format(name = name, platform = platform),
-            platform = platform,
-            zot_version = zot_version,
-        )
-
         structure_test_repositories(
             name = "{name}_st_{platform}".format(name = name, platform = platform),
             platform = platform,
             # There are already too many version attributes. No need to expose this yet.
             st_version = ST_VERSIONS.keys()[0],
         )
 
+        if zot_version:
+            zot_repositories(
+                name = "{name}_zot_{platform}".format(name = name, platform = platform),
+                platform = platform,
+                zot_version = zot_version,
+            )
+
         if register:
             native.register_toolchains("@{}//:{}_toolchain".format(crane_toolchain_name, platform))
-            native.register_toolchains("@{}//:{}_toolchain".format(zot_toolchain_name, platform))
             native.register_toolchains("@{}//:{}_toolchain".format(st_toolchain_name, platform))
+            if zot_version:
+                native.register_toolchains("@{}//:{}_toolchain".format(zot_toolchain_name, platform))
+            else:
+                native.register_toolchains("@{}//:{}_toolchain".format(crane_registry_toolchain_name, platform))
```

Comment:
maybe for another PR: the naming is confusing. crane_registry_toolchain vs crane_toolchain, seems like we named them after the tool rather than the purpose for which they are being run?

Reply:
yeah. I'll clean up the registry toolchain part to be shared across different registry types.